sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -877,30 +877,59 @@ case class GpuFromUnixTime(
   override lazy val resolved: Boolean = childrenResolved && checkInputDataTypes().isSuccess
 }
 
+
+class FromUTCTimestampExprMeta(expr: FromUTCTimestamp,
+                               override val conf: RapidsConf,
+                               override val parent: Option[RapidsMeta[_, _, _]],
+                               rule: DataFromReplacementRule)
+  extends BinaryExprMeta[FromUTCTimestamp](expr, conf, parent, rule) {
+
+  // Adding other time zones needs to carefully consider daylight saving time adjustment.
+  private val supportedUTCTimeZones = Set("UTC", "GMT")
```

Comment:
We should not hardcode the list of timezones that are supported, as there are many aliases (e.g.: Etc/UTC, Zulu).  This should leverage the Java timezone database to compare the timezone specified with what we support, e.g.:
```java
import java.time.ZoneId
...
val utc = ZoneId.of("UTC").normalized
val zone = ZoneId.of(timezone).normalized
val supported = utc.equals(zone)
```

Reply:
I didn't think of using java time utilities here. If doing so, we may have a very simple way to implement full-featured `FromUTCTimestamp`:
 * Compute time offset from the given time zone and UTC, storing in a `GpuScalar`
 * `output = add(time offset, input)`

Reply:
Ah, if time was only that simple.  Many timezones are not a fixed offset from UTC.  Simplest example is daylight savings time, where it depends on the timestamp value within its own timezone to know the offset to UTC.  There's basically a database of rules that values within a timezone need to be checked against to handle conversions.